test/functional/apps/management/_scripted_fields.js

```diff
@@ -509,10 +509,10 @@ export default function ({ getService, getPageObjects }) {
 
       it('should filter by scripted field value in Discover', async function () {
         await PageObjects.discover.clickFieldListItem(scriptedPainlessFieldName2);
-        await log.debug('filter by "Sep 17, 2015 @ 23:00" in the expanded scripted field list');
+        await log.debug('filter by "Sep 18, 2015 @ 7:52" in the expanded scripted field list');
         await PageObjects.discover.clickFieldListPlusFilter(
           scriptedPainlessFieldName2,
-          '1442531297065'
+          '1442562775953'
```

Comment:
Why has this value changed?

Reply:
I meant to call this out, but glad you did. This refactor did introduce one slight change to the behavior. When calling `Object.keys(groups)`, the values are naturally sorted, and if they all have the same document count, that natural sort will persist. In the previous implementation, whatever order the values were encountered in would be maintained in a case of a tie, with no natural sorting. From what I could tell, this seemed like just an implementation quirk, and not intended behavior.